bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitValveImpl.java

```diff
@@ -73,6 +78,73 @@ public HomekitValveImpl(HomekitTaggedItem taggedItem, List<HomekitTaggedItem> ma
         this.activeReader = new BooleanItemReader(getItem(HomekitCharacteristicType.ACTIVE_STATUS, GenericItem.class),
                 OnOffType.ON, OpenClosedType.OPEN);
         getServices().add(new ValveService(this));
+        final String timerConfig = getAccessoryConfiguration(CONFIG_TIMER, "");
+        homekitTimer = timerConfig.equalsIgnoreCase("yes") || timerConfig.equalsIgnoreCase("true");
+        if (homekitTimer) {
+            addRemainingDurationCharacteristic(taggedItem, updater);
+        }
+    }
+
+    private void addRemainingDurationCharacteristic(HomekitTaggedItem taggedItem, HomekitAccessoryUpdater updater) {
+        logger.trace("addRemainingDurationCharacteristic for {}", taggedItem);
+        ((ValveService) getPrimaryService()).addOptionalCharacteristic(new RemainingDurationCharacteristic(() -> {
```

Comment:
Why not have `ValveService` be passed in as a parameter to the `addRemainingDurationCharacteristic` function instead of having to lookup and cast it here?

Reply:
right. good point